packages/editor/src/components/post-preview-button/index.js

```diff
@@ -138,59 +94,126 @@ export class PostPreviewButton extends Component {
 
 		// Open up a Preview tab if needed. This is where we'll show the preview.
 		if ( ! this.previewWindow || this.previewWindow.closed ) {
-			this.previewWindow = window.open( '', this.getWindowTarget() );
+			this.previewWindow = window.open( '', '_blank' );
 		}
 
 		// Focus the Preview tab. This might not do anything, depending on the browser's
 		// and user's preferences.
 		// https://html.spec.whatwg.org/multipage/interaction.html#dom-window-focus
 		this.previewWindow.focus();
 
-		// If we don't need to autosave the post before previewing, then we simply
-		// load the Preview URL in the Preview tab.
-		if ( ! this.props.isAutosaveable ) {
-			this.setPreviewWindowLink( event.target.href );
-			return;
-		}
+		// Load the Preview URL in the Preview tab.
+		this.setPreviewWindowLink( event.target.href );
+	}
 
-		// Request an autosave. This happens asynchronously and causes the component
-		// to update when finished.
-		if ( this.props.isDraft ) {
-			this.props.savePost( { isPreview: true } );
-		} else {
-			this.props.autosave( { isPreview: true } );
-		}
+	setDesktopPreview() {
+		this.setState( {
+			previewSize: {
+				width: '1200px',
+				height: '800px',
+			},
+		} );
+	}
+
+	setTabletPreview() {
+		this.setState( {
+			previewSize: {
+				width: '768px',
+				height: '1024px',
+			},
+		} );
+	}
+
+	setMobilePreview() {
+		this.setState( {
+			previewSize: {
+				width: '375px',
+				height: '812px',
+			},
+		} );
+	}
 
-		// Display a 'Generating preview' message in the Preview tab while we wait for the
-		// autosave to finish.
-		writeInterstitialMessage( this.previewWindow.document );
+	closePreviewModal() {
+		this.setState( { isPreviewOpen: false } );
 	}
 
 	render() {
 		const { previewLink, currentPostLink, isSaveable } = this.props;
-
 		// Link to the `?preview=true` URL if we have it, since this lets us see
 		// changes that were autosaved since the post was last published. Otherwise,
 		// just link to the post's URL.
 		const href = previewLink || currentPostLink;
 
 		return (
-			<Button
-				isLarge
-				className="editor-post-preview"
-				href={ href }
-				target={ this.getWindowTarget() }
-				disabled={ ! isSaveable }
-				onClick={ this.openPreviewWindow }
-			>
-				{ _x( 'Preview', 'imperative verb' ) }
-				<span className="screen-reader-text">
-					{
+			<>
+				<Button
+					isLarge
+					className="editor-post-preview"
+					disabled={ ! isSaveable }
+					onClick={ this.openPreviewModal }
+				>
+					{ _x( 'Preview', 'imperative verb' ) }
+
+					<span className="screen-reader-text">
+						{
 						/* translators: accessibility text */
-						__( '(opens in a new tab)' )
-					}
-				</span>
-			</Button>
+							__( '(opens in a new tab)' )
+						}
+					</span>
+				</Button>
+				{ this.state.isPreviewOpen &&
+					<Modal
+						title={
+							// translators: Preview dialog title.
+							__( 'Preview dialog' )
```

Comment:
Please remove the word "dialog". That's already automatically announced by assistive technologies because of the role=dialog.